admin/tests-integration/udf-calc-check-status-id.spec.js

```diff
@@ -0,0 +1,134 @@
+      const checkStatusId = res[0].code
+      expect(checkStatusId).toBe('EXP') // COMPLETED
+    })
+
+    it('can identify a EXPIRED check without a pin', async () => {
+      const id = await createCheck('EXP2', 1)
+      const res = await sql.query(createQuery(id))
+      const checkStatusId = res[0].code
+      expect(checkStatusId).toBe('EXP') // COMPLETED
+    })
+
+    it('can identify a NOT RECEIVED check', async () => {
+      const id = await createCheck('NTR1', 1)
+      const res = await sql.query(createQuery(id))
+      const checkStatusId = res[0].code
+      expect(checkStatusId).toBe('NTR') // NOT RECEIVED
+    })
+
+    it('can identify a NOT RECEIVED check where the startedAt field is not set', async () => {
+      const id = await createCheck('NTR2', 1)
+      const res = await sql.query(createQuery(id))
+      const checkStatusId = res[0].code
+      expect(checkStatusId).toBe('NTR') // NOT RECEIVED
+    })
+  })
+
+  describe('try it out checks', () => {
+    it('can identify a NEW check', async () => {
+      const id = await createCheck('NEW', 0)
+      const res = await sql.query(createQuery(id))
+      const checkStatusId = res[0].code
+      expect(checkStatusId).toBe('NEW')
+    })
+
+    it('can identify a COLLECTED check', async () => {
+      const id = await createCheck('COL', 0)
+      const res = await sql.query(createQuery(id))
+      const checkStatusId = res[0].code
+      expect(checkStatusId).toBe('COL') // COLLECTED
+    })
+
+    it('can identify a STARTED check', async () => {
+      const id = await createCheck('STD', 0)
+      const res = await sql.query(createQuery(id))
+      const checkStatusId = res[0].code
+      expect(checkStatusId).toBe('STD') // STARTED
+    })
+
+    it('can identify an EXPIRED check with a pin', async () => {
+      const id = await createCheck('EXP1', 0)
+      const res = await sql.query(createQuery(id))
+      const checkStatusId = res[0].code
+      expect(checkStatusId).toBe('EXP') // COMPLETED
+    })
+
+    it('can identify an EXPIRED check without a pin', async () => {
+      const id = await createCheck('EXP2', 0)
+      const res = await sql.query(createQuery(id))
+      const checkStatusId = res[0].code
+      expect(checkStatusId).toBe('EXP') // COMPLETED
+    })
+
+    it('can identify an EXPIRED check that has been started', async () => {
+      const id = await createCheck('EXP3', 0)
+      const res = await sql.query(createQuery(id))
+      const checkStatusId = res[0].code
+      expect(checkStatusId).toBe('EXP') // COMPLETED
+    })
+
+    it('does not class a try it out check as not received', async () => {
+      const id = await createCheck('NTR1', 0)
+      const res = await sql.query(createQuery(id))
+      const checkStatusId = res[0].code
+      expect(checkStatusId).toBe('STD') // Started
+    })
+  })
+
+  describe('unusual conditions:', () => {
+    it('does not class a try it out check as complete', async () => {
```

Comment:
@GeorgeChatzigiannis @GuyHarwood - can you think of anything to add here?

Reply:
you could say not received for try out is also unusual state even though you have added the test in the block above?

Reply:
I think that's standard - all try it out checks get expired at the end of the day.  Negative tests are useful too?

Reply:
ok then the test is in the right block.